api/app/DoctrineMigrations/Version240.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version240 extends AbstractMigration
+{
+    public function getDescription() : string
+    {
+        return 'Attach all relevant lay based reports and NDRs to each lay based court order';
+    }
+
+    public function up(Schema $schema) : void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf($this->connection->getDatabasePlatform()->getName() !== 'postgresql', 'Migration can only be executed safely on \'postgresql\'.');
+
+        $this->addSql("
+update report r set court_order_id = 
+(
+	select MAX(co.id)
+	from court_order co
+	inner join court_order_deputy cod on cod.court_order_id = co.id
+	inner join dd_user u on u.id = cod.user_id
+	where r.court_order_id is null
+	and co.client_id = r.client_id
+	and u.role_name = 'ROLE_LAY_DEPUTY'
+	and u.active = true
+	group by co.client_id
+)         
+       ");
+
+        $this->addSql("
+update odr o set court_order_id = 
+(
+	select MAX(co.id)
+	from court_order co
+	inner join court_order_deputy cod on cod.court_order_id = co.id
+	inner join dd_user u on u.id = cod.user_id
+	where o.court_order_id is null
+	and co.client_id = o.client_id
+	and u.role_name = 'ROLE_LAY_DEPUTY'
+	and u.active = true
+	group by co.client_id
```

Comment:
Not sure this group by is doing anything. Not 100% sure here as can't remember how this would work but if it's just getting the max then it will just be getting the max co.id regardless. If the group by is being applied then you're getting the max for each client_id so this would bring back multiple rows and you're setting every row in odr to that co.id

Reply:
Happy to remove the group by - I have tested locally and it yields the same result as when it is not there, i.e the report.court_order_id points to the correct court order in both cases. I think I put it in as the select was bringing back more than 1 result, so my first attempt to reduce it to one was to group by, which didn't work, so I used MAX and presumably left the group by in. Nonetheless it's a great spot 👍

Reply:
Yep cool, might as well remove it then. Doesn't make much difference but to avoid confusion in future :)